docs/install/manage-add-ons-packages.md

```diff
@@ -0,0 +1,511 @@
+
+For developing add-ons for the Plone frontend, Volto, see {doc}`volto/addons/index`.
+
+
+(manage-the-problem-with-pip-label)=
+
+### The problem with pip
+
+If you want to check out a Plone core package for development, or want to override the constraints of Plone, normally you would define constraints with a file {file}`constraints.txt` to tell `pip` to install a different version of a Plone package.
+
+```
+# constraints.txt with unresolvable version conflict
+-c https://dist.plone.org/release/{PLONE_BACKEND_VERSION}/constraints.txt
+plone.api>=2.0.0a3
+```
+
+Unfortunately `pip` does not allow overriding constraints this way. 
+{term}`mxdev` solves this issue.
+
+
+(manage-mxdev-to-the-rescue-label)=
+
+### `mxdev` to the rescue!
+
+`mxdev` resolves Plone constraints with your needs for version pinning or source checkouts.
+It reads its configuration file {file}`mx.ini`, and your {file}`requirements.txt` and {file}`constraints.txt` files.
+Then it fetches the requirements and constraints of Plone
+Finally it writes new combined requirements in {file}`requirements-mxdev.txt` and new constraints in {file}`constraints-mxdev.txt`.
+Together these two files contain the combined requirements and constraints, but modified according to the configuration in {file}`mx.ini`.
+The generated files indicate from where the constraints were fetched, and comments are added when a modification was necessary.
+
+`mxdev` does not run `pip` or install packages.
+You must perform that step.
+
+
+(manage-mxdev-example-files-label)=
+
+### `mxdev` example files
+
+A minimal example set of files for `mxdev` would look like the following.
+
+{file}`requirements.txt`
+
+```ini
+-c constraints.txt
+
+Plone
+
+# List of add-ons that are needed.
+collective.easyform
+```
+
+{file}`constraints.txt`
+
+```ini
+-c https://dist.plone.org/release/{PLONE_BACKEND_VERSION}/constraints.txt
+
+# constraints of add-ons
+collective.easyform==3.4.5
+```
+
+{file}`mx.ini`
+
+```ini
+[settings]
+# constraints of Plone packages
+version-overrides =
+    plone.api>=2.0.0a3
+
+[plone.restapi]
+url = git@github.com:plone/plone.restapi.git
+branch = master
+extras = test
+```
+
+With these three files in your project, install and run `mxdev` with the following commands.
+
+```shell
+pip install mxdev
+mxdev -c mx.ini
+```
+
+`mxdev` generates the files {file}`requirements-mxdev.txt` and {file}`constraints-mxdev.txt`.
+Now you can install your packages with `pip` and the new requirements file:
+
+```shell
+pip install -r requirements-mxdev.txt
+```
+
+Finally, to reload the packages, restart your Zope instance/Plone site with the following command.
+
+```shell
+runwsgi instance/etc/zope.ini
+```
+
+```{seealso}
+This was a brief overview of how `mxdev` helps with versions and checkouts.
+It can do a lot more.
+See the [documentation of `mxdev` in its README.rst](https://github.com/mxstack/mxdev/blob/main/README.rst) for complete information.
+```
```

Comment:
Comment: Future branch of
cookiecutter-plone.starter branch https://github.com/collective/cookiecutter-plone-starter/tree/feature-4 will provide mxdev and it's configuration files like described above. 

The configuration files just need to be customized.

`make build` or `make build-backend` builds after changes.
`make start` or `make start-backen`starts rebuild instance.

Reply:
Fixed int 6c9128b1d5038f723e0b8af65fcbc2b0b9a571f0.